tests/storage/export.go

```diff
@@ -232,15 +233,9 @@ var _ = SIGDescribe("Export", func() {
 
 	createExportTokenSecret := func(name, namespace string) *k8sv1.Secret {
 		var err error
-		secret := &k8sv1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: namespace,
-				Name:      fmt.Sprintf("export-token-%s", name),
-			},
-			StringData: map[string]string{
```

Comment:
libsecret.New implementation use Data, StringData is just for convince, when the secret is created all its content get merged to the Data field.
https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/#Secret

There are places where the secret data map value are strings and in other places its bytes, I dont have strong opinion what is better, seem unimportent.

Reply:
Please reason for this in the commit message.

The API provided both options, probably to ease the user per the usage.
But now you made it harder for the user to pass the information (as a conversion is needed).
I think you need to make it easy to consume the input without making it hard to use.

You can try and use generics to make this work.

Reply:
> Please reason for this in the commit message.

Done

> The API provided both options, probably to ease the user per the usage.
> But now you made it harder for the user to pass the information (as a conversion is needed).
> I think you need to make it easy to consume the input without making it hard to use.

There are places that pass strings and other pass bytes, judging which is harder depends on usage.
I planned to change all tests to use libsecret first and do some optimizations as follow up, such as getting red of secret label for tests.

To address what you said I though about introducing `addDataEntry`, `addStringDataEntry` options to make it easier to use.
If you not insist I prefer to do in on a follow up PR.

> You can try and use generics to make this work.

I dont think generics can work in this case because the given argument (that has dynamic type) has to be assigned to a strict typed field.

Reply:
I posted https://github.com/kubevirt/kubevirt/pull/11854 .

I suggest you cherry-pick or rebase over it. Or suggest something else that does not require repeating conversions all the time.

Reply:
Done

This PR is now rebased on #11854